### PR TITLE
Improve PYTHONPATH checker

### PR DESCRIPTION
**Pull request by oxve**
Refactor to make checker work when imported and called from other module.
Also add to gn.py.

b/299098707

### CODE REVIEW

**Comment by oxve** [comment]
I've tested this locally by setting PYTHONPATH to current repo, other cobalt repo, empty and running both scripts.

**Comment by oxve** [comment]
> Is there something that prevents us from doing the same thing in gn scripts instead of in gn.py? Are we still planning to do it in gn itself as well?

It's needed in both in `gn.py` and in gn code. That's why I left the check in [BUILDCONFIG.gn](https://github.com/youtube/cobalt/blob/main/starboard/build/config/BUILDCONFIG.gn#L45) which is calling this script.

This PR modifies the python code to make it importable by `gn.py`. The previous code would not work if it was imported as `__file__` would point to the imported file in the repo that's first on `PYTHONPATH`.

**Comment by oxve** [comment]
To clarify the reasoning behind this change; A member of the the team ran into this. gn.py was failing because it imported the platforms from the wrong repo, which is before it runs `gn gen`. In that case it should assert that python modules are imported from the same path as `gn.py` is in. This is an issue for any of the python scripts we invoke directly.

On the GN side there's still the assert in `BUILDCONFIG.gn` for when `gn gen` is invoked directly.

**Comment by andrewsavage1** [comment]
> This PR modifies the python code to make it importable by `gn.py`. The previous code would not work if it was imported as `__file__` would point to the imported file in the repo that's first on `PYTHONPATH`.

Could you clarify further in what case the previous code doesn't work (maybe with an example)? I don't like the current solution because it's possible you'd want other things on your PYTHONPATH unrelated to cobalt, so I'm trying to think of a way to do it with imports like before but that doesn't have the same issue as before

**Comment by oxve** [comment]
I refactored to use the import method instead. I figured out a cleaner way to do it then what I was trying out at first.

File: starboard/build/is_on_path.py [hunk 1]
```diff
@@ -16,17 +16,31 @@
...
+    return False
 
 
 if __name__ == '__main__':
-  main()
+  try:
+    import starboard.build.is_on_path  # pylint: disable=import-outside-toplevel
+    # Use imported function instead of calling directly to ensure imports are
+    # being made from the same repo as this file is in.
+    print(str(starboard.build.is_on_path.is_on_same_path(__file__)).lower())
```
Comment (andrewsavage1) [inline starboard/build/is_on_path.py:44]: I was trying this out locally and don't think I could get it to be false—maybe you could explain it to me in person today

Comment (andrewsavage1) [reply starboard/build/is_on_path.py:44]: Actually I figured out what I did wrong, I tried it again and could get it to assert

**Review by andrewsavage1 — APPROVED** [review]

File: starboard/build/is_on_path.py [hunk 2]
```diff
@@ -16,17 +16,31 @@
...
+    True if the file is in the first entry on PYTHONPATH, False otherwise.
+  """
   try:
-    # Try to import this file and compare its path to the current file.
-    import starboard.build.is_on_path  # pylint: disable=import-outside-toplevel
-    this_file = os.path.realpath(__file__)
-    imported_file = os.path.realpath(starboard.build.is_on_path.__file__)
-    print(str(this_file == imported_file).lower())
+    return os.path.abspath(file_path).startswith(SRC_ROOT)
   except ImportError:
```
[the remark this answers is not in the document]
Comment (oxve) [reply starboard/build/is_on_path.py:35]: Removed

**Pull request merged**